lib/KSelect/index.vue

```diff
@@ -60,123 +167,606 @@
       },
-      /**
-       * Text displayed if invalid
-       */
       invalidText: {
         type: String,
         default: null,
       },
-      /**
-       * Whether or not display as inline block
-       */
-      inline: {
+      disabled: {
         type: Boolean,
         default: false,
       },
-      floatingLabel: {
+      clearable: {
         type: Boolean,
-        default: true,
+        default: false,
       },
-      placeholder: {
+      clearText: {
         type: String,
-        default: null,
+        default: '',
       },
       /**
-       * Whether to turn into a clearable state
-       * when an option has been selected.
+       * Whether or not display as inline block
        */
-      clearable: {
+      inline: {
         type: Boolean,
         default: false,
       },
-      clearText: {
-        type: String,
-        default: '',
-      },
     },
+
     data() {
       return {
+        query: '',
+        isInsideModal: false,
+        isActive: false,
+        isTouched: false,
+        highlightedOption: null,
+        showDropdown: false,
+        initialValue: JSON.stringify(this.value),
+        quickMatchString: '',
+        quickMatchTimeout: null,
+        scrollableAncestor: null,
+        dropdownButtonBottom: 'auto',
+        maxDropdownHeight: 256,
         // workaround for Keen-ui not displaying floating labels for empty objects
         selection: Object.keys(this.value || {}).length === 0 ? '' : this.value,
       };
     },
+
     computed: {
       name() {
         return `k-select-${this._uid}`;
       },
+      classes() {
+        return [
+          `ui-select-type-basic`,
+          { 'is-active': this.isActive },
+          { 'is-invalid': this.invalid },
+          { 'is-touched': this.isTouched },
+          { 'is-disabled': this.disabled },
+          { 'has-label': this.hasLabel },
+          { 'has-floating-label': this.hasFloatingLabel },
+          { 'k-select-inline': this.inline },
+          { 'k-select-disabled': this.disabled },
+        ];
+      },
+
+      labelClasses() {
+        return {
+          'is-inline': this.hasFloatingLabel && this.isLabelInline,
+          'is-floating': this.hasFloatingLabel && !this.isLabelInline,
+        };
+      },
+
+      hasLabel() {
+        return Boolean(this.label) || Boolean(this.$slots.default);
+      },
+
+      hasFloatingLabel() {
+        return this.hasLabel && this.floatingLabel;
+      },
+
+      isLabelInline() {
+        return this.selection.length === 0 && !this.isActive;
+      },
+
+      hasFeedback() {
+        return Boolean(this.invalidText) || Boolean(this.$slots.error);
+      },
+
+      showError() {
+        return this.invalid && (Boolean(this.invalidText) || Boolean(this.$slots.error));
+      },
+
+      filteredOptions() {
+        return this.options.filter((option, index) => {
+          return this.defaultFilter(option, index);
+        });
+      },
+
+      displayText() {
```

Comment:
I see some of the computed properties are extremely reduced like this one. Just a note that we should be cautious while doing this as it may break some unknown things unknowingly!

Reply:
`multiple` prop is not exposed from `KSelect`. So, I removed the logic handling of this prop. Because I think it was never used. [here](https://github.com/learningequality/kolibri-design-system/blob/develop/lib/KSelect/KeenUiSelect.vue#L396)

Reply:
yes! and this is my concern cause even tho we are not passing in the `multiple` prop directly from `Kselect` its still being used interconnectedly with other props and computed functions. from looking into the code it seems like this prop is used to indicate whether the select component can accept multiple selected options or not. Even tho this is not our current use case there is a high probability it might be needed in near future! looking it from the perspective if we keep this prop we not only make other computed properties secure but also become a little more future proof according to me! maybe @MisRob can guide us a bit more over this?

Reply:
Overall the goal was to get rid of unused logic and I think lots of APIs can be removed but it's possible that some important APIs will stay even though we don't use them right now. I would feel fine about keeping `multiple` if it feels safer. It's true this is one of the core features that may be handy.

For me, this is learning that as we continue with similar refactors, it would be good to spend some time with auditing APIs in more detail and agree beforehand what should stay and what should go. Another observation is that it'd be good to (1) move the code untouched from file to file in one commit and then (2) have a separate commit for each API removed so we can easily track dependencies and motivations for a particular API removal. I will make sure to incorporate this as suggestions to issues to come.

Thanks both, @ozer550 for your thoughtful observations and helping with review, and @Jaspreet-singh-1032 for your patience with the process and jumping with courage into a project with quite entangled tasks. This is a pioneer PR that will also help greatly with figuring out how to best approach other components.